fenwick/src/main/java/org/opencadc/fenwick/StorageSiteSync.java

```diff
@@ -0,0 +1,204 @@
+*  modify it under the terms of         modifier suivant les termes de
+*  the GNU Affero General Public        la “GNU Affero General Public
+*  License as published by the          License” telle que publiée
+*  Free Software Foundation,            par la Free Software Foundation
+*  either version 3 of the              : soit la version 3 de cette
+*  License, or (at your option)         licence, soit (à votre gré)
+*  any later version.                   toute version ultérieure.
+*
+*  OpenCADC is distributed in the       OpenCADC est distribué
+*  hope that it will be useful,         dans l’espoir qu’il vous
+*  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+*  without even the implied             GARANTIE : sans même la garantie
+*  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+*  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+*  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+*  General Public License for           Générale Publique GNU Affero
+*  more details.                        pour plus de détails.
+*
+*  You should have received             Vous devriez avoir reçu une
+*  a copy of the GNU Affero             copie de la Licence Générale
+*  General Public License along         Publique GNU Affero avec
+*  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+*  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+*                                       <http://www.gnu.org/licenses/>.
+*
+*
+************************************************************************
+*/
+
+package org.opencadc.fenwick;
+
+import ca.nrc.cadc.auth.NotAuthenticatedException;
+import ca.nrc.cadc.db.TransactionManager;
+import ca.nrc.cadc.io.ByteLimitExceededException;
+import ca.nrc.cadc.io.ResourceIterator;
+import ca.nrc.cadc.net.ResourceNotFoundException;
+import ca.nrc.cadc.net.TransientException;
+
+import java.io.IOException;
+import java.net.URI;
+import java.security.AccessControlException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.UUID;
+
+import org.apache.log4j.Logger;
+import org.opencadc.inventory.Artifact;
+import org.opencadc.inventory.InventoryUtil;
+import org.opencadc.inventory.SiteLocation;
+import org.opencadc.inventory.StorageSite;
+import org.opencadc.inventory.db.ObsoleteStorageLocation;
+import org.opencadc.inventory.db.ObsoleteStorageLocationDAO;
+import org.opencadc.inventory.db.StorageSiteDAO;
+import org.opencadc.tap.TapClient;
+
+
+/**
+ * Class to pull a single StorageSite from a remote TAP (Luskan) service.  This class is enabled by the
+ * org.opencadc.fenwick.trackSiteLocations property in the fenwick.properties file.
+ */
+public class StorageSiteSync {
+    private static final Logger log = Logger.getLogger(StorageSiteSync.class);
+
+    private static final String STORAGE_SITE_QUERY = "SELECT name, id, resourceid FROM inventory.storagesite";
```

Comment:
You also need to get the lastModified and metaChecksum columns in the query...

Reply:
OK.  Does the `Entity` class need constructor args for those values?  There is no way to set them currently.

Reply:
Nevermind, I missed the next comment.